src/main/java/com/microsoft/azure/util/AzureImdsCredentials.java

```diff
@@ -0,0 +1,49 @@
+package com.microsoft.azure.util;
+
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import com.microsoft.jenkins.azurecommons.core.credentials.TokenCredentialData;
+import hudson.Extension;
+import hudson.util.ListBoxModel;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+public class AzureImdsCredentials extends AbstractManagedIdentitiesCredentials {
+
+    public AzureImdsCredentials(CredentialsScope scope, String id, String description) {
+        super(scope, id, description);
+    }
+
+    @DataBoundConstructor
+    public AzureImdsCredentials(CredentialsScope scope, String id, String description,
+                                String azureEnvName) {
+        super(scope, id, description);
+        setAzureEnvName(azureEnvName);
+        setAzureEnvironment(AzureEnvUtil.resolveAzureEnv(azureEnvName));
+    }
+
+    @Override
+    public TokenCredentialData createToken() {
+        TokenCredentialData token = super.createToken();
+        token.setType(TokenCredentialData.TYPE_IMDS);
+        return token;
+    }
+
+    @Extension
+    public static class DescriptorImpl
+            extends BaseStandardCredentials.BaseStandardCredentialsDescriptor {
+
+        @Override
+        public String getDisplayName() {
+            return "Managed Identities for Azure Resources";
+        }
+
+        public ListBoxModel doFillAzureEnvNameItems() {
+            ListBoxModel model = new ListBoxModel();
+            model.add(AzureEnvUtil.Constants.ENV_AZURE);
+            model.add(AzureEnvUtil.Constants.ENV_AZURE_CHINA);
```

Comment:
Mooncake is supported?

Reply:
This method is copied from  AzureMsiCredentials. No changes about environment are mentioned in the documentation.